app/controllers/reactivate_account_controller.rb

```diff
@@ -1,10 +1,20 @@
 class ReactivateAccountController < ApplicationController
   include AccountReactivationConcern
+  include RememberDeviceConcern
 
   before_action :confirm_two_factor_authenticated
   before_action :confirm_password_reset_profile
 
-  def index; end
+  def index
+    @presenter = AccountShowPresenter.new(
+      decrypted_pii: nil,
+      personal_key: nil,
+      sp_session_request_url: sp_session_request_url_with_updated_params,
+      sp_name: decorated_session.sp_name,
+      decorated_user: current_user.decorate,
+      locked_for_session: pii_locked_for_session?(current_user),
+    )
```

Comment:
If we're using the presenter class just for `personal_key_generated_at`, I'd wonder if it may be better to just duplicate that snippet of code, since it's not much, `user` could be referenced from the view, and I may worry about the inconsistent state of providing some-but-not-all keyword arguments for the class constructor. Alternatively, maybe we could move the method to the `User` model class and share it in both places?

```rb
user.encrypted_recovery_code_digest_generated_at || user.active_profile&.verified_at
```

Reply:
`personal_key_generated_at` should be defined in one location, to avoid inconsistencies, drift, and duplication. We could move it to the `User` model and redefine `AccountShowPresenter` to reference that location. I would argue for a personal key service object, but there may not be enough functionality surrounding personal keys to warrant that.

Reply:
Moved to model.